contracts/SwapRouter.sol

```diff
@@ -142,10 +144,12 @@ contract SwapRouter is
                 0,
                 SwapCallbackData({
                     path: params.path.getFirstPool(), // only the first pool in the path is necessary
-                    payer: msg.sender
+                    payer: payer
                 })
             );
 
+            if (payer != address(this)) payer = address(this); // this contract pays for subsequent hops
```

Comment:
wait this is an odd way to do it, it has a branch every time plus an mstore, we could just use a ternary in the payer?

Reply:
if you just remove the if branch i'd guess it's cheaper and still correct

Reply:
well you need something like this in any case. pseudocode:

hasPaid = false
// ...
payer: hasPaid ? address(this) : msg.sender
// ...
if (!hasPaid) hasPaid = false // if statement is so we only mstore once

so this is 2 MSTOREs, and 2 conditionals / loop iteration

my version is 2 MSTOREs, and 1 conditional / loop iteration

does that seem right?

Reply:
i suspect the if statement creates a jump which is more expensive than just setting it every time

Reply:
ok, will check the gas costs and merge whichever's cheaper